src/platforms/web/runtime/directives/show.js

```diff
@@ -17,7 +17,9 @@ export default {
     if (value && transition && transition.appear && !isIE9) {
       enter(vnode)
     }
-    el.style.display = value ? '' : 'none'
+    const originalDisplay = el.style.display
+    el.style.display = value ? originalDisplay : 'none'
+    el.__vOriginalDisplay = originalDisplay
```

Comment:
flow doesn't allow attaching properties to HTMLElements, let's just declare `el`'s type as `any` in the function signatures to make it pass.